packages/arb-token-bridge-ui/src/util/withdrawals/fetchTokenWithdrawalsFromEventLogs.ts

```diff
@@ -5,42 +5,71 @@ import { Erc20Bridger } from '@arbitrum/sdk'
  * Fetches initiated token withdrawals from event logs in range of [fromBlock, toBlock].
  *
  * @param query Query params
- * @param query.fromAddress Address that initiated the withdrawal
- * @param query.toAddress Address that will receive the funds
+ * @param query.address Address that initiated the withdrawal, or received the funds
  * @param query.fromBlock Start at this block number (including)
  * @param query.toBlock Stop at this block number (including)
  * @param query.l2Provider Provider for the L2 network
  * @param query.l2GatewayAddresses L2 gateway addresses to use
  */
 export async function fetchTokenWithdrawalsFromEventLogs({
-  fromAddress,
-  toAddress,
+  sender,
+  receiver,
   fromBlock,
   toBlock,
   l2Provider,
   l2GatewayAddresses = []
 }: {
-  fromAddress?: string
-  toAddress?: string
+  sender?: string
+  receiver?: string
   fromBlock: BlockTag
   toBlock: BlockTag
   l2Provider: Provider
   l2GatewayAddresses?: string[]
 }) {
   const erc20Bridger = await Erc20Bridger.fromProvider(l2Provider)
 
-  return (
-    await Promise.all(
-      l2GatewayAddresses.map(gatewayAddress =>
+  const promises = l2GatewayAddresses.flatMap(gatewayAddress => {
+    const events = []
+
+    // funds sent by this address
+    if (sender) {
+      events.push(
         erc20Bridger.getL2WithdrawalEvents(
           l2Provider,
           gatewayAddress,
           { fromBlock, toBlock },
           undefined,
-          fromAddress,
-          toAddress
+          sender,
+          undefined
```

Comment:
this is optional, maybe just delete it?

Reply:
btw would be nice to make the accepted param in the sdk an obj in the future

Reply:
We really should, although this is a common pattern in the SDK so we will have a lot to refactor there.

Reply:
also, not a strong opinion about deleting it, but by specifying undefined it's a bit more clear in terms of what the goal here is. And it won't be accidentally added etc just because its missing.

I'm fine removing it tho if there are some strong opinions against it

Reply:
that's a good point
let's leave it